File: app/src/main/java/com/tregula/graedukacyjna/domain/database/entity/ContinentWithCountries.java
package com.tregula.graedukacyjna.domain.database.entity;

import java.util.List;

import androidx.room.Embedded;
import androidx.room.Relation;

public class ContinentWithCountries {
    @Embedded
    public Continent continent;

    @Relation(parentColumn = "id",
            entityColumn = "continentId")
    public List<Country> countries;
}

File: app/src/main/java/com/tregula/graedukacyjna/domain/database/entity/ContinentWithCountries.kt
package com.tregula.graedukacyjna.domain.database.entity

import androidx.room.Embedded
import androidx.room.Relation

class ContinentWithCountries {
    @Embedded
    var continent: Continent? = null

    @Relation(parentColumn = "id", entityColumn = "continentId")
    var countries: List<Country>? = null
}
